crates/spirv-std/src/storage_class.rs

```diff
@@ -48,6 +48,54 @@ macro_rules! storage_class {
         }
     };
 
+    // Interior Block
```

Comment:
While the `#[spirv(block)]` attribute matches our plans in this area for a hack quick fix, we were planning on having the user apply the attribute themselves to types used in block contexts. As you pointed out, there are many problems with this approach, surfacing most obviously with the storage class mismatch issues you're hitting.

(Do keep in mind that this is intended to be a quick hack fix before we're able to fully implement the bindings RFC, as that will replace this whole system with something much more elegant)